src/olympia/api/jwt_auth/views.py

```diff
@@ -24,6 +30,40 @@ class JWTKeyAuthentication(JSONWebTokenAuthentication):
     by the authenticated user.
     """
 
+    def authenticate(self, request):
+        """
+        Returns a two-tuple of `User` and token if a valid signature has been
+        supplied using JWT-based authentication.  Otherwise returns `None`.
+
+        Copied from rest_framework_jwt BaseJSONWebTokenAuthentication, with
+        the decode_handler changed to our own - because we don't want that
+        decoder to be the default one in settings.
+        """
+        jwt_value = self.get_jwt_value(request)
+        if jwt_value is None:
+            return None
+
+        try:
+            payload = handlers.jwt_decode_handler(jwt_value)
+        except jwt.ExpiredSignature:
+            log.exception('JWTKeyAuthentication signature has expired.')
```

Comment:
This should not log an exception because that will spam our Sentry. It is a user error, not an application error. I think you can just make it `log.info()`.

Also, I'd be more interested in what the actual exception message is because that's the most important thing for debugging. Something like:
````
log.info('key based JWT auth failed; it raised {exc.__class__.__name__}: {exc}'.format(exc=exc))
````
In fact, you could probably wrap all calls in a single except clause, log it, re-raise it, and catch each one as needed.